firstapp/app/src/androidTest/java/com/example/firstapp/CUJExecutor.java

```diff
@@ -0,0 +1,40 @@
+package com.example.firstapp;
+import android.os.SystemClock;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import org.junit.Test;
+
+public class CUJExecutor {
+
+  
+
+    /**
+     * Executes a CUJ with preparatory actions pre and measured actions post. Caches information for later use.
+     */
+
+    public static void executeCUJ(String[] preCUJ, String[] postCUJ) throws Exception {
+        String[] CUJ = new String[preCUJ.length + postCUJ.length];
+        if (preCUJ != null) {
+            System.arraycopy(preCUJ, 0, CUJ, 0, preCUJ.length);
+        }
+        System.arraycopy(postCUJ, 0, CUJ, preCUJ.length, postCUJ.length);
```

Comment:
Seems like your recording will include the pre actions, which is likely to be not of interest to the user.

Reply:
final recording won't include them, but we have to execute them each time we restart

Reply:
Okay, it sounds like someone will call CUJExecutor twice.